pkglistgen/tool.py

```diff
@@ -620,67 +681,98 @@ def update_and_solve_target(
 
         group = target_config.get('pkglistgen-group', '000package-groups')
         product = target_config.get('pkglistgen-product', '000product')
+        productcompose = target_config.get('pkglistgen-product', '000productcompose')
         release = target_config.get('pkglistgen-release', '000release-packages')
         oldrepos = target_config.get('pkglistgen-repos', '000update-repos')
 
-        url = api.makeurl(['source', project])
-        packages = ET.parse(http_GET(url)).getroot()
-        if packages.find('entry[@name="{}"]'.format(product)) is None:
-            if not self.dry_run:
-                undelete_package(api.apiurl, project, product, 'revive')
-            # TODO disable build.
-            logging.info('{} undeleted, skip dvd until next cycle'.format(product))
-            return
-        elif not force:
-            root = ET.fromstringlist(show_results_meta(api.apiurl, project, product,
-                                                       repository=[main_repo], multibuild=True))
-            if len(root.xpath('result[@state="building"]')) or len(root.xpath('result[@state="dirty"]')):
-                logging.info('{}/{} build in progress'.format(project, product))
-                return
-
-        drop_list = api.item_exists(project, oldrepos)
-        checkout_list = [group, product, release]
-        if drop_list and not only_release_packages:
-            checkout_list.append(oldrepos)
-
-        if packages.find('entry[@name="{}"]'.format(release)) is None:
-            if not self.dry_run:
-                undelete_package(api.apiurl, project, release, 'revive')
-            logging.info('{} undeleted, skip dvd until next cycle'.format(release))
-            return
-
         # Cache dir specific to hostname and project.
         host = urlparse(api.apiurl).hostname
         cache_dir = CacheManager.directory('pkglistgen', host, project)
 
-        if not no_checkout:
-            if os.path.exists(cache_dir):
-                shutil.rmtree(cache_dir)
-            os.makedirs(cache_dir)
+        drop_list = []
+        checkout_list = [group, product, release]
+        if git_url:
+            if os.path.exists(cache_dir + "/.git"):
+                # reset and update existing clone
+                logging.debug(subprocess.check_output(
+                    ['git', 'reset', '--hard'], cwd=cache_dir, encoding='utf-8'))
+                logging.debug(subprocess.check_output(
+                    ['git', 'pull', '--rebase'], cwd=cache_dir, encoding='utf-8'))
+            else:
+                if os.path.exists(cache_dir):
+                    shutil.rmtree(cache_dir)
+                logging.debug(subprocess.check_output(
+                    ['git', 'clone', '--recurse-submodules', git_url, cache_dir], encoding='utf-8'))
+        else:
+            url = api.makeurl(['source', project])
```

Comment:
This checks for product build state which should be independent of `git_url`

Reply:
a part below, yes, moving it up.